### PR TITLE
Release: 2023.11.10 - Documentation improvements

### PR DESCRIPTION
**Pull request by davidsmejia**
## Purpose

This PR adds:
- inaugural change log entry
- dictionary updates
- citation instruction / information
- bigger prettier file tree pictures

### Issue addressed

#155 
#171
#172 
#131
#132 
#187

## Checklist for publishing to `main`

- [X] This branch contains all commits/content ready to be released.
- [X] The most recent CHANGELOG entry includes a date.
If this release required a CHANGELOG entry, it should be today's date.

### CODE REVIEW

File: docs/download_files.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 The ScPCA Portal download packages include gene expression data, a QC report, and associated metadata for each processed sample.
 These files are delivered as a zip file.
+When you uncompress the zip file, the root directory name of your download will include the date you accessed the data on the ScPCA Portal.
+We recommend you record this date in case there are future updates to the Portal that change the underlying data or if you need to cite the data in the future (see {ref}`How to Cite <citation:how to cite>` for more information).
+<!-- TODO: Add link to CHANGELOG -->
```
Comment (jashapiro) [inline docs/download_files.md:7]: I think we need to do this before release. Copying text from `FAQ`?
```suggestion
Please see our {ref}`CHANGELOG <CHANGELOG:CHANGELOG>` for a summary of changes that impact downloads from the Portal.
```

Comment (davidsmejia) [reply docs/download_files.md:7]: ah good catch, im going to commit this to test

Comment (davidsmejia) [reply docs/download_files.md]: yup this looks good

File: .github/workflows/spell-check.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
     branches:
       - main
       - development
+      - documentation-improvements
```
Comment (jashapiro) [inline .github/workflows/spell-check.yml:11]: We can probably remove this now
```suggestion
```

**Review by jashapiro — APPROVED** [review]
LGTM!

**Pull request merged**